src/WebJobs.Script/Description/DotNet/PackageManager.cs

```diff
@@ -98,6 +104,135 @@ public static string ResolveNuGetPath()
             return path ?? NuGetFileName;
         }
 
+        public static bool RequiresPackageRestore(string functionPath)
+        {
+            string projectFilePath = Path.Combine(functionPath, DotNetConstants.ProjectFileName);
+            
+            if (!File.Exists(projectFilePath))
+            {
+                return false;
+            }
+
+            string lockFilePath = Path.Combine(functionPath, DotNetConstants.ProjectLockFileName);
+
+            if (!File.Exists(lockFilePath))
+            {
+                return true;
```

Comment:
Yeah, I was asking myself why I hadn't added that as a comment when I was typing that :)